dask_cuda/proxy_object.py

```diff
@@ -411,6 +412,7 @@ def _pxy_serialize(
         # Invalidate the (possible) cached "device_memory_objects"
         self._pxy_cache.pop("device_memory_objects", None)
 
+    @nvtx_annotate("JIT_UNSPILL", color="green", domain="dask_cuda")
```

Comment:
Yes, this is expected because of serialization I guess. Are we able to register the source of a deserialized object? I'm sure I'm ignoring the complexity of this, but maybe that could be part of the serialization/spilling metadata?

Reply:
Also, I'm fine merging this as is, but it may be worth considering whether it's useful to have deserialization of transferred data (i.e., not unspilled data) annotated just as unspilled data, I think this has the potential to be very misleading and thus cause more confusion that help.

Maybe @shwina could comment on whether this satisfies immediate needs or if we really need to think of a more sophisticated approach to prevent any potential misleading annotations.

Reply:
I agree; we ideally want to annotate _just_ unspilling (and not conflate deserialization and unspilling). Any help/direction for how we could separate those in the code would be much appreciated.

Reply:
Adding a `do-not-merge` until this conversation is resolved.

Reply:
> Notice, this is also called when the object is accessed after it has been received using TCP.

If we're definitely _not_ using the TCP transports, then is this a correct measure of data that is actually being unspilled?

If so, we could also annotate the tcp deserialize call (so you can distinguish by the nvtx call stack those unspills that are due to tcp, adn those that are not).

Reply:
> If we're definitely not using the TCP transports, then is this a correct measure of data that is actually being unspilled?

Yes, `ProxyObject` are serialized without spilling them when they are communicated using UCX

Reply:
Thanks both -- I'm afraid I still don't understand fully.

> Otherwise, we have to mark spilled ProxyObject that we are retrieving so we know not to annotate them when they are unspilled later.

At the time `ob_pxy_dask_deserialize` is called, how do we know if we're deserializing post-TCP, or if we're deserializing to unspill an object?

> we could also annotate the tcp deserialize call (so you can distinguish by the nvtx call stack those unspills that are due to tcp, adn those that are not).

This seems reasonable (we'd have to document somewhere how to interpret the annotations). Also, do you mean annotating `ProxyDetail.deserialize` specifically here?